zcash_client_sqlite/src/lib.rs

```diff
@@ -1433,11 +1671,26 @@ impl<C: BorrowMut<rusqlite::Connection>, P: consensus::Parameters> WalletWrite f
         _output: &WalletTransparentOutput,
     ) -> Result<Self::UtxoRef, Self::Error> {
         #[cfg(feature = "transparent-inputs")]
-        return wallet::transparent::put_received_transparent_utxo(
-            self.conn.borrow(),
-            &self.params,
-            _output,
-        );
+        return self.transactionally(|wdb| {
+            let (account_id, key_scope, utxo_id) =
+                wallet::transparent::put_received_transparent_utxo(
+                    wdb.conn.0,
+                    &wdb.params,
+                    _output,
+                )?;
+
+            use ReceiverRequirement::*;
+            wallet::transparent::generate_gap_addresses(
+                wdb.conn.0,
+                &wdb.params,
+                account_id,
+                key_scope,
+                &wdb.gap_limits,
+                UnifiedAddressRequest::unsafe_custom(Allow, Allow, Require),
```

Comment:
Same issue here as for https://github.com/zcash/librustzcash/pull/1673/files#r1978625962